src/anomalib/data/image/mvtec_loco.py

```diff
@@ -0,0 +1,499 @@
+      in: International Journal of Computer Vision (IJCV) 130, 947-969, 2022, DOI: 10.1007/s11263-022-01578-9
+"""
+
+import logging
+from collections.abc import Sequence
+from pathlib import Path
+
+import albumentations as A  # noqa: N812
+import cv2
+import numpy as np
+from pandas import DataFrame
+
+from anomalib import TaskType
+from anomalib.data.base import AnomalibDataModule, AnomalibDataset
+from anomalib.data.utils import (
+    DownloadInfo,
+    InputNormalizationMethod,
+    LabelName,
+    Split,
+    TestSplitMode,
+    ValSplitMode,
+    download_and_extract,
+    get_transforms,
+)
+
+logger = logging.getLogger(__name__)
+
+
+IMG_EXTENSIONS = (".png", ".PNG")
+
+DOWNLOAD_INFO = DownloadInfo(
+    name="mvtec_loco",
+    url="https://www.mydrive.ch/shares/48237/1b9106ccdfbb09a0c414bd49fe44a14a/download/430647091-1646842701"
+    "/mvtec_loco_anomaly_detection.tar.xz",
+    checksum="d40f092ac6f88433f609583c4a05f56f",
+)
+
+CATEGORIES = (
+    "breakfast_box",
+    "juice_bottle",
+    "pushpins",
+    "screw_bag",
+    "splicing_connectors",
+)
+
+GT_MERGED_DIR = "ground_truth_merged"
+
+
+def _merge_gt_mask(
+    root: str | Path,
+    extensions: Sequence[str] = IMG_EXTENSIONS,
+    gt_merged_dir: str | Path = GT_MERGED_DIR,
+) -> None:
+    """Merges ground truth masks within specified directories and saves the merged masks.
+
+    Args:
+        root (str | Path): Root directory containing the 'ground_truth' folder.
+        extensions (Sequence[str]): Allowed file extensions for ground truth masks.
+                                    Default is IMG_EXTENSIONS.
+        gt_merged_dir (str | Path]): Directory where merged masks will be saved.
+                                     Default is GT_MERGED_DIR.
+
+    Returns:
+        None
+
+    Example:
+        >>> _merge_gt_mask('path/to/breakfast_box/')
+
+        This function reads ground truth masks from the specified directories, merges them into
+        a single mask for each corresponding images (e.g. merge 059/000.png and 059/001.png into 059.png),
+        and saves the merged masks in the default GT_MERGED_DIR directory.
+
+        Note: The merged masks are saved with the same filename structure as the corresponding anomalous image files.
+    """
+    root = Path(root)
+    gt_mask_paths = {f.parent for f in root.glob("ground_truth/**/*") if f.suffix in extensions}
+
+    for mask_path in gt_mask_paths:
+        # Merge each mask inside mask_path into a single mask
+        merged_mask = None
+        for mask_file in mask_path.glob("*"):
+            if mask_file.suffix in extensions:
+                mask = cv2.imread(str(mask_file), cv2.IMREAD_UNCHANGED)
+                if merged_mask is None:
+                    merged_mask = np.zeros_like(mask)
+                merged_mask = np.maximum(merged_mask, mask)
+
+        # Binarize masks
+        merged_mask = np.minimum(merged_mask, 255)
```

Comment:
Not sure if i need to binarize the masks here or not. The ground-truth masks of the dataset is can vary, e.g., 241, 253, etc, depending on the sub-categories of the defects

Reply:
Interesting observation.  I think that is a good idea to only have 255 values in any case, since the masks are divided by 255 here:
https://github.com/openvinotoolkit/anomalib/blob/69184e378ea025d1f23ad86ebac1ada53debbe25/src/anomalib/data/base/dataset.py#L130

But I think that this line doesn't exactly do that? Isn't this element wise minimum, for example in case of 245, going to set output to 245, not 255. 
I think this should probably be `merged_mask = np.where(merged_mask > 0, 255, 0)`

Reply:
Thanks for your comment! ah yes, I think it should be `merged_mask = np.where(merged_mask > 0, 255, 0)` as you said. I have made the changes.

Reply:
Great.